realm/realm-library/src/main/cpp/io_realm_internal_OsObject.cpp

```diff
@@ -182,14 +179,13 @@ static inline size_t do_create_row_with_primary_key(JNIEnv* env, jlong shared_re
 
     if (is_pk_null) {
         if (table.find_first_null(pk_column_ndx) != npos) {
-            throw_duplicated_primary_key_exception(env, "'null'");
-            return npos;
+            throw JavaExceptionThrower(env, PK_CONSTRAINT_EXCEPTION_CLASS, format(PK_EXCEPTION_MSG_FORMAT, "'null'"));
```

Comment:
Should we add `__FILE__` and `__LINE__` (or another unique identifier) to be can easily find where the exception is thrown?